calculator.go

```diff
@@ -1,13 +1,112 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Add takes 2+ numbers and returns the result of adding them together.
+func Add(a, b float64, rest ...float64) float64 {
+	result := a + b
+	for _, item := range rest {
+		result += item
+	}
+
+	return result
+}
+
+// Subtract takes 2+ numbers and returns the result of subtracting the b
+// from the a.  For variadic args, evaluation is (a - b) - c ...
+func Subtract(a float64, b float64, rest ...float64) float64 {
+	result := a - b
+
+	for _, item := range rest {
+		result -= item
+	}
+
+	return result
+
+}
+
+// Multiply multiplies 2+ numbers.  For variadic args, evaluation is (a * b) * c ...
+func Multiply(a float64, b float64, rest ...float64) float64 {
+	result := a * b
+
+	for _, item := range rest {
+		result *= item
+	}
+
+	return result
+
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+// Divide divides 2+ numbers or returns an error if it would divide by 0.
+// For variadic args, evaluation is (a / b) / c ...
+func Divide(a float64, b float64, rest ...float64) (float64, error) {
+	if b == 0 {
+		return 0, errors.New("can't divide by zero")
+	}
+	result := a / b
+
+	for _, item := range rest {
+		if item == 0 {
+			return 0, errors.New("can't divide by zero")
+		}
+		result /= item
+	}
+
+	return result, nil
+}
+
+// Sqrt takes the square root of a nonnegative number or returns an error.
```

Comment:
"takes" is ambiguous here, since we usually use that word with functions to describe what parameters it receives. Try `"takes a nonnegative number and returns its square root..."